src/status_im/contexts/wallet/data_store_test.cljs

```diff
@@ -0,0 +1,193 @@
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds
+                                                                raw-account))}]}
+             (sut/rpc->keypair raw-keypair--private-key)))))
+
+(deftest rpc->keypairs-test
+  (testing "rpc->keypairs transforms and sorts raw keypairs"
+    (is
+     (match? [(sut/rpc->keypair raw-keypair--profile)
+              (sut/rpc->keypair raw-keypair--seed-phrase)
+              (sut/rpc->keypair raw-keypair--private-key)]
+             (sut/rpc->keypairs [raw-keypair--seed-phrase
+                                 raw-keypair--private-key
+                                 raw-keypair--profile])))))
+
+(deftest process-keypairs-test
+  (testing "process-keypairs represents updated key pairs and accounts"
+    (is
+     (match?
+      (matchers/match-with
+       [set? matchers/set-equals
+        map? matchers/equals]
+       {:removed-keypair-ids         #{}
+        :removed-account-addresses   #{}
+        :updated-accounts-by-address (->>
+                                       [raw-keypair--seed-phrase
+                                        raw-keypair--private-key]
+                                       (sut/rpc->keypairs)
+                                       (mapcat :accounts)
+                                       (utils.collection/index-by :address))
+        :updated-keypairs-by-id      {(:key-uid raw-keypair--seed-phrase)
+                                      (sut/rpc->keypair raw-keypair--seed-phrase)
+                                      (:key-uid raw-keypair--private-key)
+                                      (sut/rpc->keypair raw-keypair--private-key)}})
+      (sut/process-keypairs [raw-keypair--seed-phrase
+                             raw-keypair--private-key]))))
+  (testing "process-keypairs represents removed key pairs and accounts"
+    (is (match?
+         (matchers/match-with
+          [set? matchers/set-equals
+           map? matchers/equals]
+          {:removed-keypair-ids         #{"0x456"}
+           :removed-account-addresses   #{"1x456"}
+           :updated-accounts-by-address (->> [raw-keypair--seed-phrase]
+                                             (sut/rpc->keypairs)
+                                             (mapcat :accounts)
+                                             (utils.collection/index-by :address))
+           :updated-keypairs-by-id      {(:key-uid raw-keypair--seed-phrase)
+                                         (sut/rpc->keypair raw-keypair--seed-phrase)}})
+         (sut/process-keypairs [raw-keypair--seed-phrase
+                                (assoc raw-keypair--private-key :removed true)]))))
+  (testing "process-keypairs ignores chat accounts inside updated accounts"
+    (is
+     (match?
+      (matchers/match-with
+       [set? matchers/set-equals
+        map? matchers/equals]
```

Comment:
Just linking PRs, this [other PR](https://github.com/status-im/status-mobile/pull/20547#issuecomment-2189266291) had a few surprises related to how `match?` behaves with maps. And in the newest version of matcher-combinators there's support for `nested-equals`.

I think seeing these `match-with` in assertions repeated is something we can improve in follow-ups.

If we start to find ourselves using more and more `equals` or `nested-equals` to be more strict, it could be better to define our own test directive (also recursive), like `(is (match-strict? expected actual))`.

Reply:
yes please to `match-strict?` 🙌 

At one point while writing my tests, I was surprised my tests were fundamentally broken because I was not aware of the default matching behaviour of `match?`. So `match-strict?` would be great for this stuff.

I also wonder if there's a way to use `match?` or `match-strict?` on a subset of a map. For example, when asserting on the app-db, I would like to say that a certain set of keys is present and another set of keys is not present. Thoughts?

Reply:
> I also wonder if there's a way to use match? or match-strict? on a subset of a map.

If I understood correctly, it's a yes @seanstrom, the expected map to `match?` can use different matchers for different parts of the tree. But I don't know if the standard matchers support the use case you want. I haven't explored non-default matchers that much.